R/sim_linelist.R

```diff
@@ -0,0 +1,266 @@
+#' many iterations (internally) then the function errors, whereas if the
+#' maximum outbreak size is exceeded the function returns the data early and a
+#' warning stating how many cases and contacts are returned.
+#' @param population_age Either a `numeric` vector with two elements or a
+#' `<data.frame>` with age structure in the population. Use a `numeric` vector
+#' to specific the age range of the population, the first element is the lower
+#' bound for the age range, and and the second is the upper bound for the age
+#' range (both inclusive, i.e. \[lower, upper\]). The `<data.frame>` with
+#' age groups and the proportion of the population in that group. See details
+#' and examples for more information.
+#' @param case_type_probs A named `numeric` vector with the probability of
+#' each case type. The names of the vector must be `"suspected"`, `"probable"`,
+#' `"confirmed"`. Values of each case type must sum to one.
+#' @param config A list of settings to adjust the randomly sampled delays and
+#' Ct values. See [create_config()] for more information.
+#'
+#' @return A line list `<data.frame>`
+#' @export
+#'
+#' @author Joshua W. Lambert, Carmen Tamayo
+#'
+#' @examples
+#' # load data required to simulate line list
+#' contact_distribution <- epiparameter::epidist(
+#'   disease = "COVID-19",
+#'   epi_dist = "contact distribution",
+#'   prob_distribution = "pois",
+#'   prob_distribution_params = c(mean = 2)
+#' )
+#'
+#' infect_period <- epiparameter::epidist(
+#'   disease = "COVID-19",
+#'   epi_dist = "infectious period",
+#'   prob_distribution = "gamma",
+#'   prob_distribution_params = c(shape = 1, scale = 1)
+#' )
+#'
+#' # get onset to hospital admission from {epiparameter} database
+#' onset_to_hosp <- epiparameter::epidist_db(
+#'   disease = "COVID-19",
+#'   epi_dist = "onset to hospitalisation",
+#'   single_epidist = TRUE
+#' )
+#'
+#' # get onset to death from {epiparameter} database
+#' onset_to_death <- epiparameter::epidist_db(
+#'   disease = "COVID-19",
+#'   epi_dist = "onset to death",
+#'   single_epidist = TRUE
+#' )
+#' # example with single hospitalisation risk for entire population
+#' linelist <- sim_linelist(
+#'   contact_distribution = contact_distribution,
+#'   infect_period = infect_period,
+#'   prob_infect = 0.5,
+#'   onset_to_hosp = onset_to_hosp,
+#'   onset_to_death = onset_to_death,
+#'   hosp_risk = 0.5
+#' )
+#' head(linelist)
+#'
+#' # example with age-stratified hospitalisation risk
+#' # 20% for over 80s
+#' # 10% for under 5s
+#' # 5% for the rest
+#' age_dep_hosp_risk <- data.frame(
+#'   age_limit = c(1, 5, 80),
+#'   risk = c(0.1, 0.05, 0.2)
+#' )
+#' linelist <- sim_linelist(
+#'   contact_distribution = contact_distribution,
+#'   infect_period = infect_period,
+#'   prob_infect = 0.5,
+#'   onset_to_hosp = onset_to_hosp,
+#'   onset_to_death = onset_to_death,
+#'   hosp_risk = age_dep_hosp_risk
+#' )
+#' head(linelist)
+sim_linelist <- function(contact_distribution,
```

Comment:
Thanks for pointing this out. Initially when writing the functions I thought that it would be better to leave the most important simulation arguments (epidemiological parameters) without a default value so the user had to consider how they should parameterise the model.

However, I am not fixed to this idea and am happy to discuss what we think is the best approach for the package. I believe there have been very similar discussions for other Epiverse packages so would also be good to be consistent across the ecosystem. 

I've opened an issue (#120) from this comment to discuss this more thoroughly before making the change to the package. I will add this issue to the next release milestone (v0.4.0) so that it is not rushed to be included in the v0.3.0 release.